analyses/tmb-vaf-longitudinal/util/function-create-corplot.R

```diff
@@ -0,0 +1,66 @@
+#' Create corplots 
+#'
+#' @param maf 
+#' @param timepoints_other_plot 
+#' @param timepoints_deceased_plot 
+#' @param sid 
+#'
+#' @return A data frame with each possible combination of timepoints and create VAF corplot
+#' @export
+#'
+#' @examples
+create_corplot <- function(maf, timepoints_other_plot, timepoints_deceased_plot, sid) {
+
+  # Split maf, create df, and rename VAF column based on time point
+  timepoint_df <- maf[which(maf$timepoints_other == timepoints_other_plot), ] %>%
+    select(gene_protein, Hugo_Symbol, VAF) 
+  colnames(timepoint_df)[colnames(timepoint_df) == "VAF"] <- timepoints_other_plot
+
+  deceased_df <- maf[which(maf$timepoints_deceased == timepoints_deceased_plot), ] %>%
+    select(gene_protein, Hugo_Symbol, VAF)
+  colnames(deceased_df)[colnames(deceased_df) == "VAF"] <- timepoints_deceased_plot
+   
+  maf_join <- deceased_df %>%
+    full_join(timepoint_df, by = c("gene_protein", "Hugo_Symbol"), relationship = "many-to-many") %>%
+    mutate(sym = ifelse(Hugo_Symbol %in% oncoprint_goi$oncoprint_goi, TRUE, FALSE)) 
+  
+  maf_join[is.na(maf_join)] <- 0
+
+  # Create timepoint to use for labeling group column 
+  # only for timepoints = Diagnosis, Progressive, Recurrence
+  timepoint <- colnames(timepoint_df)
+  timepoint <- timepoint[!timepoint %in% c("gene_protein", "Hugo_Symbol")]
```

Comment:
```suggestion
  timepoint <- timepoints_other_plot
```
Can assign `timepoints` directly from `timepoints_other_plot`

Reply:
This is a good idea but it required some edit to work, see 616819e1acb321621f06b6e6d9eb6ddf77964537